### PR TITLE
Added GetGroupIDBYName and AddGroupMembers options

### PR DESCRIPTION
**Pull request by Euquimides**
Reference ticket [#7548](https://github.com/craftercms/craftercms/issues/7548)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ To create a group:
 
 `crafter-cli create-group -e local -gn test-group -gd "Description of the group"`
 
+It is possible to get a group by name:
```
Comment (sumerjabri) [inline README.md]: Remove this command.

File: src/main/groovy/org/craftercms/cli/Main.groovy [hunk 2]
```diff
@@ -18,19 +18,21 @@ package org.craftercms.cli
 
 import org.craftercms.cli.commands.AddEnvironment
 import org.craftercms.cli.commands.group.CreateGroup
+import org.craftercms.cli.commands.group.GetGroupIDByName
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/Main.groovy]: Remove this.

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+	/**
+	 * Add one or more users to a group
+	 * @param client HTTPClient object
+	 * @param groupName group name
+	 * @param users list of users to add
+	 */
+    static def addMembers(client, groupName, users) {
+        def group = GetGroupIDByName.getGroupByName(client, groupName)
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Rename `group` to `result` or `response`

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        if (!group || !group.group) {
+            println "Group not found"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Error retrieving group '`groupName`'.

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            return
+        }
+
+        def groupId = group.group.id
+        def path = "/studio/api/2/groups/${groupId}/members"
+        def body = [usernames: users]
+
+        def result = client.post(path, body)
+        if (!result) {
+            println "Failed to add members"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Failed to add members '`members`' to group '`groupName`'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            return
+        }
+
+        println "Members added successfully"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Members '`members`' added to group '`groupName`' successfully.

File: src/main/groovy/org/craftercms/cli/commands/group/GetGroupIDByName.groovy
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/GetGroupIDByName.groovy:1]: Remove this.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/groovy/org/craftercms/cli/Main.groovy [hunk 7]
```diff
@@ -18,19 +18,21 @@ package org.craftercms.cli
...
 import org.craftercms.cli.commands.user.ListUsers
+import org.craftercms.cli.commands.group.AddGroupMembers
 import picocli.CommandLine
 
 @CommandLine.Command(
         name = 'crafter-cli', usageHelpAutoWidth = true,
         versionProvider = VersionProvider.class, mixinStandardHelpOptions = true,
         subcommands = [CommandLine.HelpCommand, AddEnvironment, AddRemote, CreateSite, ListRemotes, SyncFrom, SyncTo,
-                ListSites, CopyPlugin, CreateUser, ListUsers, CreateAccessToken, PublishContent, CreateGroup, ListGroups])
+                ListSites, CopyPlugin, CreateUser, ListUsers, CreateAccessToken, PublishContent, CreateGroup, ListGroups, AddGroupMembers, GetGroupIDByName])
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/Main.groovy]: Remove GetGroupIDByName

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 8]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
+     * @param users list of users to add
+     */
     static def addMembers(client, groupName, users) {
-        def group = GetGroupIDByName.getGroupByName(client, groupName)
+        def response = getGroupIdByName(client, groupName)
 
-        if (!group || !group.group) {
-            println "Group not found"
+        if (!response || !response.group) {
+            println "Error retrieving ${groupName} group"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Error retrieving group '${groupName}'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 9]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
 
-        def groupId = group.group.id
+        def groupId = response.group.id
         def path = "/studio/api/2/groups/${groupId}/members"
         def body = [usernames: users]
 
         def result = client.post(path, body)
         if (!result) {
-            println "Failed to add members"
+            println "Failed to add ${users} to ${groupName} group"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Failed to add users '${users}' to group '${groupName}'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 10]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
             return
         }
 
-        println "Members added successfully"
+        println "The following users have been added to the ${groupName} group: ${users}"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: The users '${users}' were successfully added to group '${groupName}'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 11]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
+    /**
+     * Get a group by name
+     * @param client HTTPClient object
+     * @param groupName group name
+     */
+    static def getGroupIdByName(client, groupName) {
+        def path = "/studio/api/2/groups/by_name/${groupName}"
+        def result = client.get(path)
+        if (!result) {
+            return
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy:71]: Silent failure? Not good. We must print the error.
```
println "Failed to retrieve group '${groupName}'"

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 12]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
+        }
+
+        if (!result.group) {
+            println "Missing group with name: ${groupName}"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Failed to retrieve group '${groupName}', error '${result.message}'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 13]
```diff
@@ -31,30 +31,51 @@ class AddGroupMembers extends AbstractCommand {
...
+            return
+        }
+
+        println "Group with ${groupName} has id: ${result.group.id}"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Delete this line, it's not userful to the end user.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 14]
```diff
@@ -41,7 +41,7 @@ class AddGroupMembers extends AbstractCommand {
         def response = getGroupIdByName(client, groupName)
 
         if (!response || !response.group) {
-            println "Error retrieving ${groupName} group"
+            println "Error retrieving '${groupName}'"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: Error retrieving group '${groupName}'

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 15]
```diff
@@ -51,11 +51,11 @@ class AddGroupMembers extends AbstractCommand {
 
         def result = client.post(path, body)
         if (!result) {
-            println "Failed to add ${users} to ${groupName} group"
+            println "Failed to add '${users}' to '${groupName}'"
```
Comment (sumerjabri) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy]: This is not what I requested. Please don't resolve my comments, I'll do that. Please fix this.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please follow the comments.

File: src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy [hunk 16]
```diff
@@ -51,11 +51,11 @@ class AddGroupMembers extends AbstractCommand {
 
         def result = client.post(path, body)
         if (!result) {
-            println "Failed to add '${users}' to '${groupName}'"
+            println "Failed to add members '${users}' to group '${groupName}'"
             return
         }
 
-        println "The users '${users}' were successfully added to group '${groupName}'"
+        println "Members '${users}' added to group '${groupName}' succesfully"
```
Comment (Euquimides) [inline src/main/groovy/org/craftercms/cli/commands/group/AddGroupMembers.groovy:58]: Related with comment: https://github.com/craftercms/cli/pull/23#discussion_r1856846335

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**